plugins/org.jkiss.dbeaver.ext.kingbase/src/org/jkiss/dbeaver/ext/kingbase/model/KingbaseDatabase.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @Property(viewable = true, order = 6)
+    public String getCharacterType() {
+        return this.characterType;
+    }
+
+    @Property(viewable = true, order = 7)
+    public String getDatabaseCompatibleMode() {
+        return this.databaseCompatibleMode;
+    }
+
+
+    private void init(ResultSet dbResult) {
+        this.databaseCompatibleMode = JDBCUtils.safeGetString(dbResult, "datcompatibility");
+        this.characterType = JDBCUtils.safeGetString(dbResult, "datctype");
+    }
+
+    public void setDatabaseCompatibleMode(String databaseCompatibleMode) {
+        this.databaseCompatibleMode = databaseCompatibleMode;
+    }
+
+    public DBRProgressMonitor getMonitor() {
+        return monitor;
+    }
+
+    public void readDatabaseInfo(DBRProgressMonitor monitor) throws DBCException {
+        try (JDBCSession session = getMetaContext().openSession(monitor, DBCExecutionPurpose.META, "Load database info")) {
+            try (JDBCPreparedStatement dbStat = session
+                .prepareStatement("SELECT db.oid,db.* FROM sys_catalog.sys_database db WHERE datname=?")) {
+                dbStat.setString(1, super.getName());
+                try (JDBCResultSet dbResult = dbStat.executeQuery()) {
+                    if (dbResult.nextRow()) {
+                        init(dbResult);
+                    }
+                }
+            } catch (SQLException e) {
+                throw new DBCException(e, session.getExecutionContext());
+            }
+        }
+    }
+    
+    public Collection<PostgreLanguage> getLanguages(DBRProgressMonitor monitor) throws DBException {
+        checkInstanceConnection(monitor);
+        return languageCache.getAllObjects(monitor, this);
+    }
+    
+    @Override
+    public DBSObject refreshObject(@NotNull DBRProgressMonitor monitor) throws DBException {
+    	super.refreshObject(monitor);
+        languageCache.clearCache();
+        return this;
+    }
+    
+    public static class LanguageCache extends PostgreDatabaseJDBCObjectCache<PostgreLanguage> {
+        @NotNull
+        @Override
+        protected JDBCStatement prepareObjectsStatement(@NotNull JDBCSession session, @NotNull PostgreDatabase owner)
+            throws SQLException {
+            return session.prepareStatement(
+                "SELECT l.oid,l.* FROM sys_catalog.sys_language l " +
+                    "\nORDER BY l.oid"
+            );
+        }
+
+        @Override
+        protected PostgreLanguage fetchObject(@NotNull JDBCSession session, @NotNull PostgreDatabase owner, @NotNull JDBCResultSet dbResult)
+            throws SQLException, DBException {
+            return new PostgreLanguage(owner, dbResult);
+        }
+    }
+
+    public static class SchemaCache extends JDBCObjectLookupCache<PostgreDatabase, PostgreSchema> {
+        @NotNull
+        @Override
+        public JDBCStatement prepareLookupStatement(@NotNull JDBCSession session, @NotNull PostgreDatabase database,
```

Comment:
Please fix code style. For long parameters string put each parameter on a separate line